File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    static bool comp(vector<int>& a, vector<int>& b){
        return a[0]<b[0];
    }
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(),intervals.end(),comp);
        vector<vector<int>> ans;
        queue<vector<int>> st;
        int i=1;
        st.push(intervals[0]);
        while(i<intervals.size()){

            while(i<intervals.size() && st.back()[1]>=intervals[i][0]){
                st.back()[1] = max(st.back()[1],intervals[i][1]);
                i++;
            }
            if(i<intervals.size())
                st.push(intervals[i]);
            i++;
        }
        while(!st.empty()){
            ans.push_back(st.front());
            st.pop();
        }

        return ans;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution(object):
    def merge(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: List[List[int]]
        """
        ans = []
        intervals.sort()
        i=0
        while i<len(intervals):
            j=i+1
            while j<len(intervals) and intervals[i][1]>=intervals[j][0]:
                intervals[i][0] = min(intervals[i][0],intervals[j][0])
                intervals[i][1] = max(intervals[i][1],intervals[j][1])
                j+=1
            ans.append(intervals[i])
            i=j
        return ans
            


        